src/ORM/Association.php

```diff
@@ -424,9 +426,9 @@ public function getTarget(): Table
      *
      * @param \Closure|array $conditions list of conditions to be used
      * @see \Cake\Database\Query::where() for examples on the format of the array
-     * @return \Cake\ORM\Association
+     * @return $this
      */
-    public function setConditions($conditions)
+    public function setConditions(Closure|array $conditions)
```

Comment:
We really don't test all of these pass-through signatures.

Reply:
I am concerned that since these values get simply passed through to the corresponding query methods, people might be using values that are valid for the query but it's type not explicitly listed in the association methods.

Reply:
I meant that I agree. We don't seem to test all the paths to the query methods.

Reply:
Updated Association::setConditions() and BelongsToMany::setConditions()